### ISSUE TITLE
Zowe 2.3.1: ZWESLSTC ends with RC=8 (same error as with 2.3.0)

### ISSUE THREAD

**Issue by Hermann63** [zowe/community#1706]
**Describe the bug**
ZWESLSTC 2.3.1 (and same as with 2.3.0) **does not start** with the same /etc/zowe2.yaml as Zowe 2.2 was started with before.

**Steps to Reproduce**
S ZWESLSTC

**Expected behavior**
Should start with the same /etc/zowe2.yaml as Zowe 2.2 was started successfully before !

**Screenshots**
                      J E S 2  J O B  L O G  --  S Y S T E M  S H 0 H  --  N O D E  C H M V S H          
  
 12.36.13 STC04334 ---- MONDAY,    03 OCT 2022 ----
 12.36.13 STC04334  IEF695I START ZWESLSTC WITH JOBNAME ZWESLSTC IS ASSIGNED TO USER STCZOWE , GROUP STCGRP
 12.36.13 STC04334  $HASP373 ZWESLSTC STARTED
 12.36.13 STC04334  IEF403I ZWESLSTC - STARTED - TIME=12.36.13
 12.36.15 STC04334  GSDMV20I JOBNAME- --STEP-- -PSTEP-- CCODE ELAPSED-TIME -CPU-TIME-
 12.36.15 STC04334  GSDMV21I ZWESLSTC STARTING ZWELNCH      8     00:00:02      0.02S
 12.36.15 STC04334  IEF404I ZWESLSTC - ENDED - TIME=12.36.15
 12.36.15 STC04334  GSDMV22I -ZWESLSTC ENDED.  NAME-* NO NAME PROVIDED * TOTAL CPU TIME=   .00  TOTAL ELAPSED TIME=   .04
 12.36.15 STC04334  $HASP395 ZWESLSTC ENDED - RC=0008                            

**Logs**
 2022-10-03 10:36:13 <ZWELNCH:83951784> STCZOWE INFO ZWEL0021I Zowe Launcher starting
 2022-10-03 10:36:13 <ZWELNCH:83951784> STCZOWE INFO ZWEL0023I Zowe YAML config file is '/etc/zowe2.yaml'
 2022-10-03 10:36:13 <ZWELNCH:83951784> STCZOWE INFO ZWEL0024I HA_INSTANCE_ID is '{{ha_instance_id}}'
 2022-10-03 10:36:13 <ZWELNCH:83951784> STCZOWE INFO ZWEL0020I loading '/etc/zowe2.yaml'
 2022-10-03 10:36:13 <ZWELNCH:83951784> STCZOWE INFO ZWEL0017I ROOT_DIR is '/ZOWE2'
 2022-10-03 10:36:15 <ZWELNCH:83951784> STCZOWE WARN ZWEL0066W command '/ZOWE2/bin/zwe internal config get --configmgr --config "/etc/zowe2.yaml" --ha-instance {{ha_instance_id}} --path .zowe.workspaceDirectory' ended with code 256
 2022-10-03 10:36:15 <ZWELNCH:83951784> STCZOWE ERROR ZWEL0059E failed to get WORKSPACE_DIR dir
 2022-10-03 10:36:15 <ZWELNCH:83951784> STCZOWE ERROR ZWEL0032E failed to find WORKSPACE_DIR='      missing required property 'pem' at '/zowe/certificate'', check if the dir exists

**Details**
 - Version and build number: 2.3.1 (PTFs UO02053 and UO02054 applied successfully)
 - Test environment: Parallel SYSPLEX SH00, z/OS Rev 2.4 RSU level 2208, RACF)
 - started with empty /var/zowe/workspace and /var/zowe/logs

**Web Browser Details (if the bug relates to Zowe Desktop usage):**
 - not applicable at that stage

**REST API client (in case of REST API issue):**
 - not applicable at that stage

**Shell Environment Details (if the bug relates to CLI):**
 - not applicable at that stage

**Additional context**

see above, JESLOG and SYSPPRINT attached 
[ZWESLSTC.ZOWE231.JESLOB.TXT](https://github.com/zowe/community/files/9696895/ZWESLSTC.ZOWE231.JESLOB.TXT)
[ZWESLSTC.ZOWE231.SYSPRINT.TXT](https://github.com/zowe/community/files/9696896/ZWESLSTC.ZOWE231.SYSPRINT.TXT)

startup config zowe2.yaml added as TXT
[zowe2.TXT](https://github.com/zowe/community/files/9697380/zowe2.TXT)

**Comment by skurnevich** [comment]
Hello Hermann,

Zowe v2.3.0 with the _configmgr_ mode enabled enforces validation of the _zowe.yaml_ configuration against the _zowe.yaml_ _schema_, that is what you see in the output. 
It says that something in your config is not matching the schema - specifically "missing required property 'pem' at '/zowe/certificate'"

You are right that **pem** section looks redundant when you use a _keyring_ instead of a _keystore_, but **zowe.certificate.pem.certificateAuthorities** value is still required according to the schema, otherwise, we should update our schema for this field to be optional.

One question - was this **zowe.certificate** section generated automatically (**--update-config** option for **zwe init** command) or had you updated this section manually in the yaml file afterward? 

As a solution, could you please try to add **pem** section to your **zowe.certificate** and fill in the **certificateAuthorities** value to point to your CA in the keyring? Other values could be empty, it should look something like that

```
    pem:
      key: 
      certificate: 
      certificateAuthorities: safkeyring:////STCZOWE/ZoweKeyring2&<your-CA>
```

**Comment by Hermann63** [comment]
Hi Sergei,

thanks for feedback. 

I started with the sampled provided example-zowe.yaml and tried to update all related to RACF. # # >>>> Certificate setup scenario 4
During the error I searched looked into PEM related stuff but found only examples with USS keystore, therefore PEM statements were not in my zowe2.yaml .

After searching during last days,...found this: 

**./zwe init certificate -c /etc/zowe2.yaml --update-config --security-dry-run**  

This now generates PEM statements in **zowe.certificate** section:

pem:                                                 
  key: '""'                                          
  certificate: '""'                                  
  certificateAuthorities: >-                         
    safkeyring:////STCZOWE/ZoweKeyring2&ZoweLocalCAV2
    
With this, ZWESLSTC is now passing this hurdle.

Might be a good idea to provide easier and more sample-yamls for USS and different RACF sections in the future ? 
...or add this kind of errors into the troubleshooting section on Zowe Doc ?

I am pretty sure that others might fall into this trap as well...then information about this might be helpful for them as well.

regards and thanks for your help
Hermann
